crates/runtime-config/src/lib.rs

```diff
@@ -46,82 +49,85 @@ pub struct ResolvedRuntimeConfig<T> {
     pub toml: toml::Table,
 }
 
+impl<T> ResolvedRuntimeConfig<T> {
+    pub fn summarize(&self, runtime_config_path: Option<&Path>) {
```

Comment:
Moving this code here seems a bit odd. Is this crate intended to be tightly coupled to the Spin CLI?

If it stays here, maybe rename to something like `print_summary` to make its purpose more clear?

Reply:
Yea I agree - though the inlining it is also not great. If someone wants to create an alternate `RuntimeFactorsBuilder` and still print out this summary (which I would expect since they're already going to get all other logging for free), they'll have to copy this code from the `spin-runtime-factors` crate.

Reply:
Thinking about this more - I think as the crate stands today, this is the best place for this. This has a high chance to change though as we start tweaking things.